tutorials/gui_for_python_applications/README.md

```diff
@@ -0,0 +1,187 @@
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.setupUi()  # Setup the UI
+        self.runHoloscanApp()  # Run the Holoscan application
+```
+
+At a high level, this is all we need to launch a Python Holoscan application with a GUI.
+The `setupUi()` method defines the GUI widgets and layout, while `runHoloscanApp()` runs the Florence-2 application in a separate thread.
+Details of these methods are explored in the following sections.
+
+## Creating the GUI Widgets and Layout
+
+The `setupUi()` method creates the GUI with a few simple widgets using PySide6 APIs. 
+For those unfamiliar with PySide6, this [tutorial](https://www.pythonguis.com/pyside6-tutorial/) provides an introduction.
+
+```python
+    def setupUi(self):
+        """Setup the UI components."""
+        self.setWindowTitle("Florence-2")
+        self.resize(400, 150)
+        self.centralWidget = QWidget()
+        self.setCentralWidget(self.centralWidget)
+
+        layout = QVBoxLayout()
+
+        # Create and add dropdown for task selection
+        self.dropdown = QComboBox()
+        self.dropdown.addItems(
+            [
+                "Object Detection",
+                "Caption",
+                "Detailed Caption",
+                "More Detailed Caption",
+                "Dense Region Caption",
+                "Region Proposal",
+                "Caption to Phrase Grounding",
+                "Referring Expression Segmentation",
+                "Open Vocabulary Detection",
+                "OCR",
+                "OCR with Region",
+            ]
+        )
+        layout.addWidget(QLabel("Select an option:"))
+        layout.addWidget(self.dropdown)
+
+        # Create and add text input for prompt
+        self.text_input = QLineEdit()
+        layout.addWidget(QLabel("Enter text:"))
+        layout.addWidget(self.text_input)
+
+        # Create and add submit button
+        self.submit_button = QPushButton("Submit")
+        self.submit_button.clicked.connect(self.on_submit)
+        layout.addWidget(self.submit_button)
+
+        self.centralWidget.setLayout(layout)
+```
+
+This code creates the following widgets:
+
+* <b>Drop-down Menu</b>: Lists the vision tasks supported by Florence-2.
+* <b>Text input Widget</b>: Allows text input for tasks such as Open Vocabulary Detection.
+* <b>Submit Button</b>: Triggers the `on_submit()` method when clicked.
+
+When the application is running, the user selects a vision task, enters text (if 
+needed), and clicks "Submit" to change the task performed by the model.
+The `on_submit()` method is then invoked, calling the `set_parameters()` method 
+in the `FlorenceApp` class to update the operator's parameters.
+
+```python
+    def on_submit(self):
+        """Handle the submit button click event."""
+        selected_option = self.dropdown.currentText()
+        entered_text = self.text_input.text()
+
+        # Set parameters in the Holoscan application
+        global gApp
```

Comment:
Why do we need a `global` here?

Reply:
global is needed here so that gApp is referring to the global var and defining a local variable.  If we remove this statement, the global gApp remains as None and the on_submit() function will not do anything